### DISCUSSION TITLE
Is the servo project actively encouraging non-expert contributions?

### DISCUSSION THREAD

**Discussion by crispinb** [servo/servo#29573] [General] [answered]
I know the Servo project in its earlier iteration had quite a reputation for being accessible to contributors who aren't necessarily Rust experts.  But I know a lot's changed organisationally, and the project has grown. So is it still worthwhile for a relative Rust neophyte to poke around and look for something to tackle?

**Comment by Nighink** [comment]
There are efforts to make things better for new contributors so I suppose everybody new is welcome. Servo has been dormant for a bit and now has people that work on it again. See the [blog posts](https://servo.org/blog/).
I haven't contributed anything yet, can't even get it to compile right now, so don't take my word for it. I suppose everybody welcomes somebody who is interested. So go for it and ask for help if you encounter problems!

**Comment by crispinb** [comment]
_everybody welcomes somebody who is interested_

Unfortunately that isn't always true. There are plenty of repos where you can practically hear the groans from contributors when someone new pokes their head in. Sometimes there are good reasons for this - eg. complex specialised codebases with poor docs/comments, and no-one available with time to explain to newcomers. But in any case it's good to know if that's the case before investing time.

Servo apparently (just from what I read) did cultivate a welcoming attitude. I was just wondering if this is still the case. Perhaps it's too large and complex for this now (case in point your compilation problems!).

**Comment by tsurumi-yizhou** [comment]
I'm either new to servo. In my opinion, we can start from reading code and write documents. As I know servo project hasn't a doc website.

**Comment by crispinb** [comment]
I'll take the lack of response as a 'no'.

**Comment by mukilan** [comment, marked as answer]
Hi @crispinb, 

Sorry about the delayed response. Servo project is indeed open to contributions, but as we are currently in the initial stages of restarting the project and the near-term focus is on the layout engine, we don't have too many tasks where we can provide active mentoring for beginners.

However, we do welcome contributions that require some guidance including pull request reviews, discussions and proposals for new features etc.

Hope this provides some clarity about the current state of the project.
